proto/substrait/algebra.proto

```diff
@@ -123,6 +124,23 @@ message ReadRel {
     substrait.extensions.AdvancedExtension advanced_extension = 10;
   }
 
+  // Read an Iceberg Table
+  message IcebergTable {
+    oneof table_type {
+      MetadataFileRead direct = 1;
+      // future: add catalog table types (e.g. rest api, latest metadata in path, etc)
+    }
+
+    // Read an Iceberg table using a metadata file
+    message MetadataFileRead {
+      // the specific uri of a metadata file (e.g. s3://mybucket/mytable/<ver>-<uuid>.metadata.json)
+      string metadata_uri = 1;
+
+      // the snapshot id to read. if not set, uses the current snapshot listed in the metadata file
+      optional string snapshot_id = 2;
```

Comment:
Typically we have the choice of specifying snapshot or timestamp. Is the intention that the engine will know enough to derive snapshot if given a timestamp